### DISCUSSION TITLE
Spawning a framebuffer app from inside koreader

### DISCUSSION THREAD

**Discussion by enthdegree** [koreader/koreader#12935] [Q&A] [answered]
Hello, on a Kobo Clara BW I would like to spawn fbpad from inside koreader to go back and forth between the two. fbpad draws fine when run from nickel. But when fbpad is run from koreader its draws are corrupt: all black, screen rotation with `echo [1|2|3|4] > /sys/class/graphics/fb0` makes a mess.

Koreader must have reconfigured the framebuffer. Is it obvious to anyone what's happening? 

fbpad's drawing is structurally simple: it reads the framebuffer info structs to get memory locations, then draws a terminal there. I patched fbpad to init a copy of fbink and ask for a full refresh (zero-filled settings struct) every time a char is drawn.

**Comment by Frenzie** [comment, marked as answer]
https://github.com/koreader/koreader/blob/15f5dbef83b84beb5ef3ddd8c21fc6b7f7c615fc/platform/kobo/koreader.sh#L320-L371

**Reply by enthdegree** [reply]
Thanks!! Will try soon

**Comment by NiLuJe** [comment]
Don't rotate the FB manually, most ntx boards will do weird shit ;).

Go through fbdepth or fbink_set_fb_info

**Comment by enthdegree** [comment]
Thanks, indeed it was the bit depth and my bad rotation
